components/script/serviceworker_manager.rs

```diff
@@ -87,6 +106,11 @@ struct ServiceWorkerRegistration {
     waiting_worker: Option<ServiceWorker>,
     /// https://w3c.github.io/ServiceWorker/#dfn-installing-worker
     installing_worker: Option<ServiceWorker>,
+    /// A channel to send control message to the worker,
+    /// currently only used to signal shutdown.
+    control_sender: Option<Sender<()>>,
+    /// A handle to join on the worker thread.
+    join_handle: Option<JoinHandle<()>>,
```

Comment:
There are two places in this PR where we have optional shutdown signal, optional Thread handle, and we do “take then join” in Drop. If we wrap JoinHandle into a type which joins on Drop, than we can remove options and custom code from Drop (and one custom Drop impl altogether) and replace then with `// filed(drop) order is significant` comment.

https://github.com/rust-analyzer/ra_vfs/blob/c4a2aaaf3c2d5231b11ee21a5f6ca34c3955805c/src/io.rs#L55

Reply:
Thanks, noted, I think that's something I might look into if we end-up doing this in more places. Personally I'm not sure relying on comments and field order is necessarily more robust than some additional code like with those drop implementations...

Reply:
Yeah, I also prefer to write something than to rely on implicit drop order. What swayed me for VFS case is that I was able to remove the Option(which unfortunately infected code other than Drop). If OptIon is required anyway, manual impl indeed would be better.